### PR TITLE
fix: what I have observed, almost all typo

### PR DESCRIPTION
**Pull request by dev-itsheng**
# PR Details

Recently, I tried to understand front-end engineering, read the code of Modern.js, learned a lot, and fixed some problems I found.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [ ] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dev-itsheng** [comment]
> Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please follow the instructions at https://bytedance.feishu.cn/docs/doccnDUWkuuoOzFGtzoeMuWpzKc

I don't have permission to this linked document (probably because I'm not a ByteDance employee yet), please tell me how to sign the CLA, thanks.

**Comment by chenjiahan** [comment]
Thanks for your PR, we are looking at the CLA issue, please wait for our feedback.

Also, the `Lint` and `Unit Test` CI are failed, please fix the CI.

**Comment by chenjiahan** [comment]
@dev-itsheng We have updated the CLA bot, please sign online at [this link](https://cla-assistant.io/modern-js-dev/modern.js?pullRequest=983).

**Comment by dev-itsheng** [comment]
> Thanks for your PR, we are looking at the CLA issue, please wait for our feedback.
> 
> Also, the `Lint` and `Unit Test` CI are failed, please fix the CI.

Due to time reasons, I only unit-tested the places I modified, not all of the unit tests in the project, so I didn't notice that there was a test that didn't belong to my changes that failed.

It goes like this:

image

source code:

```typescript
describe('plugin-nocode', () => {
  it('expect add style-loader to webpack config in nocode plugin', () => {
    try {
      userConfig._raw = userConfig;
      userConfig.source.include = ['query-string'];
      const config = new BaseWebpackConfig(
        appContext as IAppContext,
        userConfig,
      ).config();

      const hasStyleLoader = config?.module?.rules?.[1]?.oneOf.findIndex(
        (rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
            rule?.use?.[0]?.loader === 'style-loader'
          );
        },
      );
      expect(hasStyleLoader).toBeLessThan(0);
      handleWebpackConfig(config, {
        isDev: true,
        appDirectory: appContext.appDirectory,
        umdEntryFile: './',
      });

      const hasStyleLoaderAfterHandled =
        config?.module?.rules?.[1]?.oneOf.findIndex((rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
            rule.use[0].loader === 'style-loader'
          );
        });

      // ---------------------------------------------
      // this line, `hasStyleLoaderAfterHandled` is -1
      // ---------------------------------------------
      expect(hasStyleLoaderAfterHandled).toBeGreaterThanOrEqual(0);
    } catch (e) {
      console.error(e);
    }
  });
});
```

I looked at the changes related to this file and found that there was indeed one critical change that affected this test, see https://github.com/modern-js-dev/modern.js/pull/969, in `packages/cli/plugin-nocode/src/compiler/index.ts`, he change `'style-loader'` to `require.resolve('style-loader')`, but not modify the unit test:

```diff
  (webpackConfig?.module?.rules as RuleSetRule[])?.[1]?.oneOf?.forEach(rule => {
    if (
      Array.isArray(rule.use) &&
      typeof rule.use[0] === 'object' &&
      (rule.use[0]?.loader || '').includes('mini-css-extract-plugin')
    ) {
-      rule.use[0].loader = 'style-loader';
+      rule.use[0].loader = require.resolve('style-loader');
    }
  });
```

I have modified the unit test and made it pass:

```diff
describe('plugin-nocode', () => {
  it('expect add style-loader to webpack config in nocode plugin', () => {
    try {
      userConfig._raw = userConfig;
      userConfig.source.include = ['query-string'];
      const config = new BaseWebpackConfig(
        appContext as IAppContext,
        userConfig,
      ).config();

      const hasStyleLoader = config?.module?.rules?.[1]?.oneOf.findIndex(
        (rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
-           rule?.use?.[0]?.loader === 'style-loader'
+           rule?.use?.[0]?.loader === require.resolve('style-loader')
          );
        },
      );
      expect(hasStyleLoader).toBeLessThan(0);
      handleWebpackConfig(config, {
        isDev: true,
        appDirectory: appContext.appDirectory,
        umdEntryFile: './',
      });

      const hasStyleLoaderAfterHandled =
        config?.module?.rules?.[1]?.oneOf.findIndex((rule: RuleSetRule) => {
          return (
            Array.isArray(rule.use) &&
            typeof rule.use[0] === 'object' &&
-           rule?.use?.[0]?.loader === 'style-loader'
+           rule?.use?.[0]?.loader === require.resolve('style-loader')
          );
        });
      expect(hasStyleLoaderAfterHandled).toBeGreaterThanOrEqual(0);
    } catch (e) {
      console.error(e);
    }
  });
});
```

**Comment by chenjiahan** [comment]
Yeah, this case is failed due to #969, thanks for you fix~

**Comment by dev-itsheng** [comment]
I looked at some Lint errors, tried to fix some, others unsure turned off with `eslint-disable`

File: packages/review/testing/src/config/testConfigOperator.ts [hunk 1]
```diff
@@ -31,22 +31,19 @@ class TestConfigOperator {
     return this._testConfig;
   }
 
-  public mergeJestConfig(commingConfig: JestConfig) {
-    this._jestConfig = merge({}, this._jestConfig, commingConfig);
+  public mergeJestConfig(commonConfig: JestConfig) {
```
Comment (chenjiahan) [inline packages/review/testing/src/config/testConfigOperator.ts:34]: I think `commingConfig` is not a typo, it means `newConfig`

File: website/i18n/zh-cn/docusaurus-theme-classic/footer.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
   },
   "link.item.label.专题详解": {
     "message": "专题详解",
-    "description": "The label of footer link with label=专题详解 linking to /docs/guides/features/runtime/code-spliting"
+    "description": "The label of footer link with label=专题详解 linking to /docs/guides/features/runtime/code-splitting"
```
Comment (chenjiahan) [inline website/i18n/zh-cn/docusaurus-theme-classic/footer.json:28]: The `code-spliting.md` should be renamed too.

File: packages/review/testing/src/config/testConfigOperator.ts (hunk 1 above)
Comment (dev-itsheng) [reply packages/review/testing/src/config/testConfigOperator.ts:34]: So it's actually `comingConfig`.

But I still find the word weird, not even `newConfig` is appropriate, maybe we should use `userConfig`?

File: website/i18n/zh-cn/docusaurus-theme-classic/footer.json (hunk 2 above)
Comment (dev-itsheng) [reply website/i18n/zh-cn/docusaurus-theme-classic/footer.json:28]: Yes, I renamed it.

File: packages/review/testing/src/config/testConfigOperator.ts (hunk 1 above)
Comment (dev-itsheng) [reply packages/review/testing/src/config/testConfigOperator.ts:34]: After thinking about it, if I were to design a `merge`, I would call the incoming new parameter `source` and the original parameter `target`.

Comment (chenjiahan) [reply packages/review/testing/src/config/testConfigOperator.ts:34]: `source` and `target` is fine

File: packages/cli/core/compiled/ajv-keywords/ajv/dist/types/jtd-schema.d.ts [hunk 3]
```diff
@@ -34,7 +34,7 @@ export declare type SomeJTDSchemaType = (// ref
 declare type RequiredKeys<T> = {
     [K in keyof T]-?: undefined extends T[K] ? never : K;
 }[keyof T];
-/** optional or undifined-able keys of an object */
+/** optional or undefined-able keys of an object */
```
Comment (chenjiahan) [inline packages/cli/core/compiled/ajv-keywords/ajv/dist/types/jtd-schema.d.ts:37]: `compiled` directory contains the code of pre-bundled dependencies, we should not manually modify the code in this directory.

Comment (dev-itsheng) [reply packages/cli/core/compiled/ajv-keywords/ajv/dist/types/jtd-schema.d.ts]: Yes, I reversed the changes to the three files in the `compiled` directory.

File: packages/review/testing/src/config/testConfigOperator.ts (hunk 1 above)
Comment (dev-itsheng) [reply packages/review/testing/src/config/testConfigOperator.ts:34]: OK, I change `commonConfig` to `sourceConfig`.

**Comment by chenjiahan** [comment]
There are some ESLint issues in old code that cause CI to fail, this Pull Request will be merged first and the ESLint issues will be fixed later.

**Pull request merged**